File: android/src/main/kotlin/com/affise/attribution/affise_attribution_lib/ext/TouchTypeExt.kt
package com.affise.attribution.affise_attribution_lib.ext

import com.affise.attribution.events.predefined.TouchType

fun String.toTouchType(): TouchType? {
    return when (this) {
        "CLICK" -> TouchType.CLICK
        "WEB_TO_APP_AUTO_REDIRECT" -> TouchType.WEB_TO_APP_AUTO_REDIRECT
        "IMPRESSION" -> TouchType.IMPRESSION
        else -> null
    }
}

File: ios/Classes/extensions/TouchTypeExt.swift
import AffiseAttributionLib


internal func toTouchType(_ value: String?) -> TouchType? {
    switch value {
    case "CLICK": return TouchType.CLICK
    case "WEB_TO_APP_AUTO_REDIRECT": return TouchType.WEB_TO_APP_AUTO_REDIRECT
    case "IMPRESSION": return TouchType.IMPRESSION
            
    default:
        return nil
    }
}
